src/components/intent.ts

```diff
@@ -241,15 +250,38 @@ export class Intent {
     }
 
     /**
-     * Return the client this Intent is acting on behalf of.
+     * Legacy property to access the matrix-js-sdk.
+     * @deprecated Support for the matrix-js-sdk client will be going away in
+     * a future release. Where possible, the intent object functions should be
+     * used. The `botSdkIntent` also provides access to the new client.
+     * @see getClient
+     */
+    public get client() {
+        return this.getClient();
+    }
+
+    /**
+     * Return a matrix-js-sdk client, which is created on demand.
+     * @deprecated Support for the matrix-js-sdk client will be going away in
+     * a future release. Where possible, the intent object functions should be
+     * used. The `botSdkIntent` also provides access to the new client.
      * @return The client
      */
     public getClient() {
-        return this.client;
+        if (this.legacyClient) {
+            return this.legacyClient;
+        }
+        if (!this.opts.getJsSdkClient) {
+            throw Error('Legacy client not available');
+        }
+        log.warn("Support for the matrix-js-sdk will be going away in a future release." +
+        "Please update occurances of Intent.getClient() and Intent.client");
```

Comment:
And change them to what? Not use the client directly?
This message lacks an explanation of what to do instead,

Reply:
I've:

- Only made the warning fire once
- Made the warning fire via the `console`, so it is never missed by developers.
- Explained how to replace the calls.